tests/test_model_name.py

```diff
@@ -0,0 +1,110 @@
+# TODO: Unit test get_model_name
+from tests.compat import assert_outcomes
+
+
+def test_generic_model_with_custom_name_no_warning(testdir):
+    testdir.makepyfile(
+        """
```

Comment:
NIP: let's indent whole string content or just this line to make code more readable